sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/aggregate/GeneralAggregateFunc.java

```diff
@@ -60,4 +63,18 @@ public GeneralAggregateFunc(String name, boolean isDistinct, Expression[] childr
 
   @Override
   public Expression[] children() { return children; }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    GeneralAggregateFunc that = (GeneralAggregateFunc) o;
+    return Objects.equals(name, that.name) && Objects.equals(isDistinct, that.isDistinct) &&
```

Comment:
bool primitive comparison works faster than `Objects.equals`, also it is usually placed at the beginning, because it is faster to check than string/objects.
```suggestion
    return isDistinct == that.isDistinct && Objects.equals(name, that.name) &&
```

Reply:
Yeah. bool primitive comparison faster than Objects.equals. But put bool primitive comparison at first is not good for readability.